lua/telescope/algos/linked_list.lua

```diff
@@ -215,4 +215,25 @@ function LinkedList:ipairs()
   end
 end
 
+function LinkedList:truncate(max_results)
```

Comment:
Hmm, we could decide whether to go from the front or the end based on the length and the max results.

For example, if we have 500,000 results and we limit 700,000 then there's actually no reason to go at all.

If we have 1,000,001 results and limit is 1,000,000 better to just truncate the last entry.